### PR TITLE
blog: subpath

### PR DESCRIPTION
**Pull request by lieut-data**
This describes the new subpath feature for developers.

This blog isn't yet "mainstream", but this feels like the right place for a "point in time" description of a feature.

### CODE REVIEW

File: site/content/blog/subpath.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+---
+title: "Subpath Support"
+date: 2018-06-25T15:35:09-04:00
+draft: true
+---
+Mattermost 5.1 will include support for [serving Mattermost from subpaths](https://mattermost.atlassian.net/browse/MM-10366). This allows Mattermost to be exposed at something like https://example.com/company/mattermost, with your proxy server exposing different services at other subpaths. Subpath support is configured via the [Site URL](https://docs.mattermost.com/administration/config-settings.html?highlight=siteurl#site-url). 
+
+#### Using subpaths in production
+
+In production, after configuration and restart, the application server will expect all HTTP requests to be anchored to the configured subpath. This includes static assets, API calls, and websockets. For convenience, the application server will redirect any unexpected requests back into this subpath, e.g. forwarding http://example.com/login to http://example.com/company/mattermost/login.
```
Comment (jasonblais) [inline site/content/blog/subpath.md]: `will expect` --> `expects`

Comment (jasonblais) [reply site/content/blog/subpath.md]: `websockets` --> `WebSockets` or `WebSocket events`

Comment (jasonblais) [reply site/content/blog/subpath.md]: `will redirect` --> `redirects`

File: site/content/blog/subpath.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+The pre-built static assets shipped with Mattermost assume being hosted at `/static/`. In production, when the `SiteURL` is configured with a subpath such as https://example.com/company/mattermost, the application server will rewrite any affected assets on startup to point at `/company/mattermost/static` instead. This rewriting can be done manually with the `mattermost config subpath` command:
```
Comment (jasonblais) [inline site/content/blog/subpath.md]: `will rewrite` --> `rewrites`

File: site/content/blog/subpath.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+make run
+```
+
+To resume testing at the root, either define a `SiteURL` with no subpath or unset the environment variable:
+```
+unset MM_SERVICESETTINGS_SITEURL
+make restart
+```
+
+Note that this environment variable is also honoured by the server as an override to config.json as described in [Configuration Settings](https://docs.mattermost.com/administration/config-settings.html#configuration-settings), so feel free to throw this export in your shell's initialization script to apply to both the server and the webapp simultaneously.
```
Comment (jasonblais) [inline site/content/blog/subpath.md]: Propose 

```
This environment variable is also honoured by the server as an override to config.json as described in [documentation](https://docs.mattermost.com/administration/config-settings.html#configuration-settings). Therefore, you can include this export in your shell's initialization script to apply to both the server and the webapp simultaneously.
```

^wondering if `shell's initialization script` is clear to non-English devs? 0/5

File: site/content/blog/subpath.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+#### How does it work?
```
Comment (jasonblais) [inline site/content/blog/subpath.md]: `How do subpaths work?`

File: site/content/blog/subpath.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+Under the covers, in development, our [webpack.config.js](https://github.com/mattermost/mattermost-webapp/blob/daefd1c76844612f3aaccfb5e42f12000e59cbfd/webpack.config.js#L136) is parsing the `MM_SERVICESETTINGS_SITEURL` environment variable to extract the subpath. This is first used by [HtmlWebpackPlugin](https://webpack.js.org/plugins/html-webpack-plugin/) to anchor the assets loaded by [root.html](https://github.com/mattermost/mattermost-webapp/blob/master/root.html). It is then exported into the node environment as `process.env.PUBLIC_PATH` and used by [entry.js](https://github.com/mattermost/mattermost-webapp/blob/master/entry.js) to define [`__webpack_public_path__`](https://webpack.js.org/guides/public-path/), affecting all modules loaded dynamically after the web application starts.
```
Comment (jasonblais) [inline site/content/blog/subpath.md]: `is parsing` --> `parses`

File: site/content/blog/subpath.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
+---
+title: "Subpath Support"
+date: 2018-06-25T15:35:09-04:00
+draft: true
+---
+Mattermost 5.1 will include support for [serving Mattermost from subpaths](https://mattermost.atlassian.net/browse/MM-10366). This allows Mattermost to be exposed at something like https://example.com/company/mattermost, with your proxy server exposing different services at other subpaths. Subpath support is configured via the [Site URL](https://docs.mattermost.com/administration/config-settings.html?highlight=siteurl#site-url).
```
[the remark this answers is not in the document]
Comment (lieut-data) [reply site/content/blog/subpath.md]: Yeah, the hope was to pre-announce and make it developer focussed in this case (the JIRA in turn links to other resources like Twitter requests and UserVoice, which might be relevant.)

File: site/content/blog/subpath.md (hunk 3 above)
Comment (lieut-data) [reply site/content/blog/subpath.md]: Proposal sounds good. I think `shell` and `initialization script` are standard dev terms for `.bashrc` and `.zshrc` and the like, which would probably be familiar to non-English devs. I'm clearly using the force here :)

**Comment by lieut-data** [comment]
Great feedback! I've pushed changes :)

**Pull request merged**